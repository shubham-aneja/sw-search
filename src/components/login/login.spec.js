import React from 'react';
import { mount } from 'enzyme';

import Login from './index.js';

describe('Login Component', () => {
    it('rendering and handler testing', () => {
        const onLoginMock = jest.fn();
        const onUserNameChangeMock = jest.fn();
        const onPasswordChangeMock = jest.fn();
        const loginDestroyMock = jest.fn();


        const props = {
            error: '',
            onLogin: onLoginMock,
            onPasswordChange: onPasswordChangeMock,
            onUsernameChange: onUserNameChangeMock,
            password: 'abcd',
            username: 'Luke',
            isLoading: false,
            loginDestroy: loginDestroyMock
        };
        const login = mount(<Login {...props} />)
        expect(login.html()).toMatchSnapshot();

        expect(onUserNameChangeMock).not.toBeCalled()
        expect(onPasswordChangeMock).not.toBeCalled()

        const inputs = login.find('input')
        const userNameInput = inputs.at(0)
        userNameInput.simulate('change', { target: { value: 'Luke S' }})

        const passwordInput = inputs.at(1)
        passwordInput.simulate('change', { target: { value: 'abcd' }})

        expect(onUserNameChangeMock).toBeCalled();
        expect(onPasswordChangeMock).toBeCalled();

        const button = login.find('.action-button');
        button.simulate('click');
        login.setProps({ isLoading: true });
        button.simulate('click');
        login.setProps({ error: 'unit test case error' });
    });
});

