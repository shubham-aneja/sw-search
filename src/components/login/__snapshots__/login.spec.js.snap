// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component rendering and handler testing 1`] = `"<div class=\\"home-app\\"><h2 class=\\"welcome-element\\">Welcome to login component</h2><div class=\\"fields-section\\"><div class=\\"field-container\\"><div class=\\"field-label\\">Username</div><div class=\\"field-editor\\"><input type=\\"text\\" value=\\"Luke\\"></div></div><div class=\\"field-container\\"><div class=\\"field-label\\">Password</div><div class=\\"field-editor\\"><input type=\\"password\\" value=\\"abcd\\"></div></div></div><div class=\\"info-section\\"><div class=\\"info-item\\"><span>Username and password are case sensitive</span></div><div class=\\"info-item\\"><span>For testing: use Luke Skywalker and 19BBY</span></div><div class=\\"info-item\\"><span>For limited calls testing: use C-3PO and 112BBY</span></div></div><div class=\\"actions-section\\"><div class=\\"action-button\\">Login</div></div></div>"`;
